providers/efa/verbs.c

```diff
@@ -88,6 +88,51 @@ int efa_query_device_ex(struct ibv_context *context,
 	return 0;
 }
 
+int efadv_query_device(struct ibv_context *ibvctx,
+		       struct efadv_device_attr *attr)
+{
+	struct efa_context *ctx = to_efa_context(ibvctx);
+	struct efa_dev *dev = to_efa_dev(ibvctx->device);
+	uint64_t comp_mask_out = 0;
+
+	if (!is_efa_dev(ibvctx->device))
+		return EOPNOTSUPP;
+
+	attr->version = 0;
+	attr->flags = 0;
+
+	if ((attr->comp_mask & EFADV_DEVICE_ATTR_MAX_SQ_WR) && dev->max_sq_wr) {
```

Comment:
Don't you need to check that user-supplied valid comp_mask?

Reply:
What do you mean by valid comp_mask?
The user fills whatever he wants as comp_mask input, if he filled something that's not supported then it will not be set on comp_mask output.

Reply:
I see that you did it as in mlx5.

Reply:
It should not be so fined grained in the initial version, just use 0.

Reply:
Use 0 as what?

Reply:
@jgunthorpe ?

Reply:
0 as the input comp_mask. The first iteration of the struct should have a 0 comp_mask for uses like this.

Reply:
Done